src/components/PostDetails.tsx

```diff
@@ -2,105 +1,92 @@
+import React, { useEffect, useState } from 'react';
+import * as commentService from '../api/comments';
 import { Loader } from './Loader';
+import { CommentList } from './CommentList';
 import { NewCommentForm } from './NewCommentForm';
+import { Post } from '../types/Post';
+
+type Props = {
+  selestedPost: Post;
+};
+
+export const PostDetails: React.FC<Props> = ({ selestedPost }) => {
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [isCommentsLoading, setIsCommentsLoading] = useState(false);
+  const [hasError, setHasError] = useState(false);
+  const [isCommentFormShown, setIsCommentFormShown] = useState(false);
+
+  useEffect(() => {
+    setIsCommentsLoading(true);
+    commentService
+      .getComments(selestedPost.id)
+      .then(setComments)
+      .catch(() => setHasError(true))
+      .finally(() => setIsCommentsLoading(false));
+  }, [selestedPost]);
+
+  function deleteComment(commId: number) {
+    setComments(currComments => currComments.filter(com => commId !== com.id));
+
+    return commentService.deleteComment(commId).catch(error => {
+      setComments(comments);
+      setHasError(true);
+      throw error;
+    });
+  }
 
-export const PostDetails: React.FC = () => {
   return (
     <div className="content" data-cy="PostDetails">
       <div className="content" data-cy="PostDetails">
         <div className="block">
           <h2 data-cy="PostTitle">
-            #18: voluptate et itaque vero tempora molestiae
+            #{selestedPost.id}: {selestedPost.title}
```

Comment:
The typo in `selestedPost` should be corrected to `selectedPost` to match the prop name.